src/XAD/Jacobian.hpp

```diff
@@ -175,5 +178,16 @@ class Jacobian
     std::vector<T> v_;
     unsigned int domain_, codomain_;
     std::vector<std::vector<T>> matrix_;
+
+    template <typename U>
+    static auto has_begin_impl(int) -> decltype(std::declval<U>().begin(), std::true_type{});
+
+    template <typename U>
+    static std::false_type has_begin_impl(...);
+
+    template <typename U>
+    struct has_begin : decltype(has_begin_impl<U>(0))
+    {
+    };
```

Comment:
Nice - I think it's time to introduce some kind of helpers or utils header for these.

Reply:
For now I put these in Hessian.hpp. Should I make a new header file for these? If so what should it be named and does it warrant its own tests?